TST: 3.4.0: pytest is failing in `test/test_sync.py::test_get_transform_grid_list__contains` unit

I'm packaging your module as an rpm package so I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w --no-isolation`
- because I'm calling `build` with `--no-isolation` I'm using during all processes only locally installed modules
- install .whl file in </install/prefix>
- run pytest with PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>

Here is pytest output:
<details>

```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-pyproj-3.4.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-pyproj-3.4.0-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra --import-mode=importlib
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.13, pytest-7.1.3, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/pyproj-3.4.0, configfile: pytest.ini
plugins: mock-3.8.2, anyio-3.6.1
collected 840 items

test/test_aoi.py .........................                                                                                                                           [  2%]
test/test_awips221.py .                                                                                                                                              [  3%]
test/test_cli.py ....................................                                                                                                                [  7%]
test/test_database.py ...................................................                                                                                            [ 13%]
test/test_datadir.py ......................                                                                                                                          [ 16%]
test/test_datum.py ..                                                                                                                                                [ 16%]
test/test_datum_shift.py ....                                                                                                                                        [ 16%]
test/test_doctest_wrapper.py ..                                                                                                                                      [ 17%]
test/test_exception_logging.py ..                                                                                                                                    [ 17%]
test/test_geod.py .......sssssssssssssssssssss.................                                                                                                      [ 22%]
test/test_list.py ....                                                                                                                                               [ 23%]
test/test_network.py ...........                                                                                                                                     [ 24%]
test/test_pickle.py .                                                                                                                                                [ 24%]
test/test_proj.py ......................................                                                                                                             [ 29%]
test/test_show_versions.py ....                                                                                                                                      [ 29%]
test/test_sync.py ....F.........                                                                                                                                     [ 31%]
test/test_transform.py .                                                                                                                                             [ 31%]
test/test_transformer.py ..s.....................................................................................................................................    [ 47%]
test/test_utils.py ..............                                                                                                                                    [ 49%]
test/crs/test_crs.py ............................................................................................................................................... [ 66%]
.............................................................................................                                                                        [ 77%]
test/crs/test_crs_cf.py ..................................................                                                                                           [ 83%]
test/crs/test_crs_coordinate_operation.py ....................................................                                                                       [ 89%]
test/crs/test_crs_coordinate_system.py ............................                                                                                                  [ 92%]
test/crs/test_crs_datum.py .......                                                                                                                                   [ 93%]
test/crs/test_crs_json.py ...............................                                                                                                            [ 97%]
test/crs/test_crs_maker.py .......................                                                                                                                   [100%]

================================================================================= FAILURES =================================================================================
__________________________________________________________________ test_get_transform_grid_list__contains __________________________________________________________________

    @pytest.mark.network
    def test_get_transform_grid_list__contains():
        grids = get_transform_grid_list(
            bbox=BBox(170, -90, -170, 90),
            spatial_test="contains",
            include_already_downloaded=True,
        )
        assert len(grids) > 5
        source_ids = set()
        for grid in grids:
            source_ids.add(grid["properties"]["source_id"])
>       assert sorted(source_ids) == ["no_kv", "nz_linz"]
E       AssertionError: assert ['nz_linz'] == ['no_kv', 'nz_linz']
E         At index 0 diff: 'nz_linz' != 'no_kv'
E         Right contains one more item: 'nz_linz'
E         Use -v to get more diff

test/test_sync.py:85: AssertionError
============================================================================= warnings summary =============================================================================
test/test_datadir.py::test_get_data_dir__from_user[str]
test/test_datadir.py::test_get_data_dir__from_user[Path]
test/test_datadir.py::test_append_data_dir__internal[str]
test/test_datadir.py::test_append_data_dir__internal[Path]
  /home/tkloczko/rpmbuild/BUILDROOT/python-pyproj-3.4.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/pyproj/datadir.py:38: UserWarning: pyproj unable to set database path.
    _global_context_set_data_dir()

test/test_transformer.py::test_transformer_group__unavailable
test/test_transformer.py::test_transformer_group__network_disabled
test/test_transformer.py::test_transformer_group__download_grids
test/test_transformer.py::test_transformer_group__download_grids__directory
  /home/tkloczko/rpmbuild/BUILDROOT/python-pyproj-3.4.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/pyproj/transformer.py:197: UserWarning: Best transformation is not available due to missing Grid(short_name=us_noaa_alaska.tif, full_name=, package_name=, url=https://cdn.proj.org/us_noaa_alaska.tif, direct_download=True, open_license=True, available=False)
    super().__init__(

test/test_transformer.py::test_transformer_group__get_transform_crs
  /home/tkloczko/rpmbuild/BUILDROOT/python-pyproj-3.4.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/pyproj/transformer.py:197: UserWarning: Best transformation is not available due to missing Grid(short_name=nl_nsgi_rdtrans2018.tif, full_name=, package_name=, url=https://cdn.proj.org/nl_nsgi_rdtrans2018.tif, direct_download=True, open_license=True, available=False)
    super().__init__(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================= short test summary info ==========================================================================
SKIPPED [1] test/test_geod.py:359: Missing shapely
SKIPPED [1] test/test_geod.py:365: Missing shapely
SKIPPED [1] test/test_geod.py:375: Missing shapely
SKIPPED [1] test/test_geod.py:393: Missing shapely
SKIPPED [1] test/test_geod.py:405: Missing shapely
SKIPPED [1] test/test_geod.py:417: Missing shapely
SKIPPED [1] test/test_geod.py:438: Missing shapely
SKIPPED [1] test/test_geod.py:449: Missing shapely
SKIPPED [1] test/test_geod.py:468: Missing shapely
SKIPPED [1] test/test_geod.py:479: Missing shapely
SKIPPED [1] test/test_geod.py:487: Missing shapely
SKIPPED [1] test/test_geod.py:493: Missing shapely
SKIPPED [1] test/test_geod.py:503: Missing shapely
SKIPPED [1] test/test_geod.py:521: Missing shapely
SKIPPED [1] test/test_geod.py:533: Missing shapely
SKIPPED [1] test/test_geod.py:545: Missing shapely
SKIPPED [1] test/test_geod.py:566: Missing shapely
SKIPPED [1] test/test_geod.py:588: Missing shapely
SKIPPED [1] test/test_geod.py:599: Missing shapely
SKIPPED [1] test/test_geod.py:618: Missing shapely
SKIPPED [1] test/test_geod.py:629: Missing shapely
SKIPPED [1] test/test_transformer.py:56: https://github.com/OSGeo/PROJ/issues/2425
FAILED test/test_sync.py::test_get_transform_grid_list__contains - AssertionError: assert ['nz_linz'] == ['no_kv', 'nz_linz']
========================================================== 1 failed, 817 passed, 22 skipped, 9 warnings in 27.38s ==========================================================
```
</details>

Here is list of installed modules in build env
<details>

```console
Package                       Version
----------------------------- -----------------
alabaster                     0.7.12
anyio                         3.6.1
appdirs                       1.4.4
argon2-cffi                   21.3.0
argon2-cffi-bindings          21.2.0
asttokens                     2.0.5
attrs                         22.1.0
Babel                         2.10.2
backcall                      0.2.0
beautifulsoup4                4.11.1
bleach                        5.0.0
branca                        0.5.0
Brlapi                        0.8.3
build                         0.8.0
CacheControl                  0.12.11
cachy                         0.3.0
certifi                       2021.10.8
cffi                          1.15.1
charset-normalizer            2.1.0
cleo                          0.8.1
clikit                        0.6.2
cloudpickle                   2.2.0
codespell                     2.1.0
crashtest                     0.3.1
cryptography                  36.0.2
cssselect                     1.1.0
Cython                        0.29.32
debugpy                       1.6.3
decorator                     5.1.1
defusedxml                    0.7.1
deprecation                   2.1.0
distlib                       0.3.6
distro                        1.7.0
docutils                      0.18.1
doit                          0.36.0
entrypoints                   0.4
executing                     0.8.3
extras                        1.0.0
fastjsonschema                2.16.1
filelock                      3.8.0
fixtures                      4.0.0
gpg                           1.17.1-unknown
html2text                     2020.1.16
html5lib                      1.1
httpretty                     1.1.4
idna                          3.3
imagesize                     1.4.1
importlib-metadata            4.12.0
importlib-resources           5.9.0
iniconfig                     1.1.1
ipykernel                     6.15.3
ipython                       8.4.0
ipython-genutils              0.2.0
ipywidgets                    8.0.1
jaraco.classes                3.2.2
jedi                          0.18.1
jeepney                       0.8.0
Jinja2                        3.1.1
json5                         0.9.9
jsonschema                    4.16.0
jupyter_client                7.3.5
jupyter_core                  4.11.1
jupyter_packaging             0.12.3
jupyter-server                1.18.1
jupyter-sphinx                0.4.0
jupyterlab                    3.4.7
jupyterlab-pygments           0.1.2
jupyterlab_server             2.15.1
jupyterlab-widgets            3.0.2
keyring                       23.9.1
libcomps                      0.1.19
lockfile                      0.12.2
louis                         3.23.0
lxml                          4.9.1
MarkupSafe                    2.1.1
matplotlib-inline             0.1.5
metaextract                   1.0.8
mistune                       2.0.3
more-itertools                8.13.0
msgpack                       1.0.4
nbclassic                     0.4.2
nbclient                      0.6.8
nbconvert                     7.0.0
nbformat                      5.5.0
nest-asyncio                  1.5.5
notebook                      6.4.12
notebook-shim                 0.1.0
numpy                         1.23.1
packaging                     21.3
pandas                        1.4.2
pandocfilters                 1.5.0
parso                         0.8.3
pastel                        0.2.1
pbr                           5.9.0
pep517                        0.12.0
pexpect                       4.8.0
pickleshare                   0.7.5
pip                           22.2.2
pkginfo                       1.8.3
pkgutil_resolve_name          1.3.10
platformdirs                  2.5.2
pluggy                        1.0.0
ply                           3.11
poetry-core                   1.2.0
prometheus-client             0.14.1
prompt-toolkit                3.0.29
psutil                        5.9.2
ptyprocess                    0.7.0
pure-eval                     0.2.2
py                            1.11.0
py2pack                       0.8.7
pycparser                     2.21
Pygments                      2.13.0
PyGObject                     3.42.2
pylev                         1.4.0
pyparsing                     3.0.9
pypi-search                   1.2.1
pyrsistent                    0.18.1
pytest                        7.1.3
pytest-mock                   3.8.2
python-dateutil               2.8.2
pytz                          2022.1
pyzmq                         23.2.0
requests                      2.28.1
requests-toolbelt             0.9.1
rpm                           4.17.0
scour                         0.38.2
SecretStorage                 3.3.2
Send2Trash                    1.8.0
setuptools                    65.3.0
shellingham                   1.5.0
six                           1.16.0
sniffio                       1.2.0
snowballstemmer               2.2.0
soupsieve                     2.3.2.post1
Sphinx                        5.1.1
sphinx-argparse               0.3.1
sphinxcontrib-applehelp       1.0.2.dev20220730
sphinxcontrib-devhelp         1.0.2.dev20220730
sphinxcontrib-htmlhelp        2.0.0
sphinxcontrib-jsmath          1.0.1.dev20220730
sphinxcontrib-qthelp          1.0.3.dev20220730
sphinxcontrib-serializinghtml 1.1.5
stack-data                    0.5.0
terminado                     0.15.0
testtools                     2.5.0
tinycss2                      1.1.1
tomli                         2.0.1
tomlkit                       0.11.4
tornado                       6.2
traitlets                     5.4.0
urllib3                       1.26.12
virtualenv                    20.16.4
wcwidth                       0.2.5
webencodings                  0.5.1
websocket-client              1.4.1
wheel                         0.37.1
widgetsnbextension            4.0.3
xarray                        2022.6.0
xyzservices                   2022.6.0
zipp                          3.8.1
```
</details>

--- a/test/test_sync.py
+++ b/test/test_sync.py
@@ -82,7 +82,7 @@ def test_get_transform_grid_list__contains():
     source_ids = set()
     for grid in grids:
         source_ids.add(grid["properties"]["source_id"])
-    assert sorted(source_ids) == ["no_kv", "nz_linz"]
+    assert sorted(source_ids) == ["nz_linz"]
 
 
 @pytest.mark.network

